configure.py

```diff
@@ -0,0 +1,140 @@
+    if not os.path.exists(os.path.join(iree_dialects_path, "CMakeLists.txt")):
+      print(f"ERROR: Cannot find iree-dialects project at {iree_dialects_path}")
+      return 1
+    # Must come before the sandbox project.
+    llvm_projects.insert(0, "iree_dialects")
+    #llvm_projects.append("iree_dialects")
+    llvm_configure_args.append(
+        f"-DLLVM_EXTERNAL_IREE_DIALECTS_SOURCE_DIR={iree_dialects_path}")
+
+  # Detect LLVM.
+  if args.llvm_path:
+    llvm_path = os.path.abspath(args.llvm_path)
+    print(f"-- Using explicit llvm-project path: {llvm_path}")
+  elif llvm_path:
+    print(f"-- Using inferred llvm-project path: {llvm_path}")
+  else:
+    llvm_path = os.path.join(args.repo_root, "..", "llvm-project")
+    print(f"-- Using default llvm-project path: {llvm_path}")
+  if not os.path.exists(llvm_path):
+    print(f"ERROR: Could not find llvm-project at {llvm_path}")
+    return 1
+
+  # Detect clang.
+  clang_path = shutil.which("clang")
+  clangpp_path = shutil.which("clang++")
+  has_clang = False
+  if clang_path and clangpp_path:
+    llvm_configure_args.append(f"-DCMAKE_C_COMPILER={clang_path}")
+    llvm_configure_args.append(f"-DCMAKE_CXX_COMPILER={clangpp_path}")
+    has_clang = True
+  else:
+    print(
+        "WARNING: Could not find clang. Building with default system compiler")
+
+  # Detect lld.
+  lld_path = shutil.which("ld.lld")
+  if not lld_path:
+    print("WARNING: LLD (ld.lld) not found on path. Configure may fail.")
+
+  # Detect ccache.
+  ccache_path = shutil.which("ccache")
+  if args.enable_ccache:
+    if ccache_path:
+      print(f"-- Using ccache")
+      llvm_configure_args.append("-DLLVM_CCACHE_BUILD=ON")
+    else:
+      print("WARNING: Project developers use ccache which is not installed")
+
+  # CMake configure.
+  build_dir = os.path.abspath(args.build_dir)
+  os.makedirs(build_dir, exist_ok=True)
+  cmake_args = [
+      "cmake",
+      "-GNinja",
+      f"-B{build_dir}",
+      f"-S{os.path.join(llvm_path, 'llvm')}",
+      "-DLLVM_ENABLE_LLD=ON",
+      f"-DLLVM_ENABLE_PROJECTS={';'.join(llvm_builtin_projects)}",
+      "-DLLVM_TARGETS_TO_BUILD=X86;NVPTX",
+      "-DMLIR_INCLUDE_INTEGRATION_TESTS=ON",
+      "-DLLVM_ENABLE_ASSERTIONS=ON",
+      "-DLLVM_INCLUDE_UTILS=ON",
+      "-DLLVM_INSTALL_UTILS=ON",
+      "-DLLVM_BUILD_EXAMPLES=ON",
+      "-DMLIR_ENABLE_BINDINGS_PYTHON=ON",
+      f"-DPython3_EXECUTABLE={sys.executable}",
+      "-DCMAKE_BUILD_TYPE=Release",
+      f"-DLLVM_EXTERNAL_PROJECTS={';'.join(llvm_projects)}",
+  ] + llvm_configure_args
+  print(f"-- Running cmake:\n  {' '.join(cmake_args)}")
+  subprocess.check_call(cmake_args, cwd=build_dir)
+
+  # Write out .env.
+  with open(f"{os.path.join(args.repo_root, '.env')}", "wt") as f:
+    f.write(f"PYTHONPATH={os.path.join(build_dir, 'tools', 'sandbox', 'python_package')}")
+
+  # Do initial build.
+  cmake_args = ["cmake", "--build", build_dir, "--target", "tools/sandbox/all"]
```

Comment:
I am not clear where the target `tools/sandbox/all` comes from? 
I had to do a bunch of changes to the setup to work on my machine, atm it is failing the iree build:
https://github.com/google/iree-llvm-sandbox/pull/17

I'll try one more time and if it does not go through I'll have to temporarily disable it until you make another pass at it.

Reply:
Not a problem to disable.

That is just cmake/ninja shortcuts. You will find tools/sandbox as a directory in your build dir. And cmake generates an "all" target for each directory. Building subsets this way is usually how i work in large projects.

Llvm puts all external projects under "tools". That is where mlir is too. So tools/mlir/all works as well.